[ios] fix runtime python3

fuzzard opened the pull request:
## Description
_posixsubprocess.c cant be not compiled, as the python os module automatically adds
subprocess that requires _posixsubprocess.c. This is a change in how it was done in python2.7

## Motivation and Context
Runtime crashes with python3

## How Has This Been Tested?
Tested with youtube plugin on ipad.

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

fuzzard commented:
Please note, there are further changes required that i havent gotten around to yet.
Error from service.xbmc.versioncheck script regarding waitpid still needs to be dealt with.

```
>Python callback/script returned the following error<--
 - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
Error Type: <class 'AttributeError'>
Error Contents: module 'os' has no attribute 'waitpid'
Traceback (most recent call last):
  File "/private/var/containers/Bundle/Application/E777F838-70C2-49C5-9034-B77D704159D5/Kodi.app/AppData/AppHome/addons/service.xbmc.versioncheck/service.py", line 20, in <module>
    import platform
  File "/var/containers/Bundle/Application/E777F838-70C2-49C5-9034-B77D704159D5/Kodi.app/Frameworks/lib/python3.7/platform.py", line 116, in <module>
    import sys, os, re, subprocess
  File "/var/containers/Bundle/Application/E777F838-70C2-49C5-9034-B77D704159D5/Kodi.app/Frameworks/lib/python3.7/subprocess.py", line 608, in <module>
    class Popen(object):
  File "/var/containers/Bundle/Application/E777F838-70C2-49C5-9034-B77D704159D5/Kodi.app/Frameworks/lib/python3.7/subprocess.py", line 1548, in Popen
    def _internal_poll(self, _deadstate=None, _waitpid=os.waitpid,
AttributeError: module 'os' has no attribute 'waitpid'
-->End of Python script error report<--
```

If anyone can test other addons on ios, and let me know of any other errors. I dont use addons in my day to day usage, so not sure whats best to test everything.

Rechi commented on tools/depends/target/python3/Makefile:
```diff
@@ -41,7 +41,8 @@ $(PLATFORM): $(TARBALLS_LOCATION)/$(ARCHIVE) $(DEPS)
 	cd $(PLATFORM); patch -p1 -i ../android-nl_langinfo.patch
 	cd $(PLATFORM); patch -p1 -i ../apple.patch
 ifeq ($(OS),darwin_embedded)
-	cd $(PLATFORM); patch -p1 -i ../make-fork-optional.patch
```
As that file is now unused also remove it from the git repo.

Rechi commented on tools/depends/target/python3/Makefile:
```diff
@@ -41,7 +41,8 @@ $(PLATFORM): $(TARBALLS_LOCATION)/$(ARCHIVE) $(DEPS)
 ifeq ($(OS),darwin_embedded)
-	cd $(PLATFORM); patch -p1 -i ../make-fork-optional.patch
+	cd $(PLATFORM); patch -p1 -i ../darwin_embedded.patch
+	cd $(PLATFORM); sed -ie 's|MACHDEP="unknown"|MACHDEP="darwin"|' configure.ac
```
What's this needed for?

fuzzard replied:
we actually use the darwin platform as a code path. ios isnt detected as a platform (because it isnt in python), so we can either piggyback off darwin, or we implement the ios platform and then ifdefs go nuts through the code base

Rechi replied:
Please check if it is really needed in the current state of feature_python3 branch.

fuzzard replied:
ok, running up a build now

fuzzard replied:
seems ok from my limited testing. Have pushed update

Rechi commented on tools/depends/target/python3/darwin_embedded.patch:
```diff
@@ -0,0 +1,276 @@
+diff -ru Python-3.7.4_original/Lib/os.py Python-3.7.4/Lib/os.py
```
remove

Rechi commented on tools/depends/target/python3/darwin_embedded.patch:
```diff
@@ -0,0 +1,276 @@
+diff -ru Python-3.7.4_original/Lib/os.py Python-3.7.4/Lib/os.py
+--- Python-3.7.4_original/Lib/os.py	2019-07-09 04:03:50.000000000 +1000
```
```suggestion
--- a/Lib/os.py
```

Rechi commented on tools/depends/target/python3/darwin_embedded.patch:
```diff
@@ -0,0 +1,276 @@
+diff -ru Python-3.7.4_original/Lib/os.py Python-3.7.4/Lib/os.py
+--- Python-3.7.4_original/Lib/os.py	2019-07-09 04:03:50.000000000 +1000
++++ Python-3.7.4/Lib/os.py	2019-08-21 10:58:34.000000000 +1000
```
```suggestion
+++ b/Lib/os.py
```

Rechi commented on tools/depends/target/python3/darwin_embedded.patch:
```diff
@@ -0,0 +1,276 @@
+         except (FileNotFoundError, NotADirectoryError) as e:
+             last_exc = e
+         except OSError as e:
+diff -ru Python-3.7.4_original/Lib/platform.py Python-3.7.4/Lib/platform.py
```
remove

Rechi commented on tools/depends/target/python3/darwin_embedded.patch:
```diff
@@ -0,0 +1,276 @@
+             last_exc = e
+         except OSError as e:
+diff -ru Python-3.7.4_original/Lib/platform.py Python-3.7.4/Lib/platform.py
+--- Python-3.7.4_original/Lib/platform.py	2019-07-09 04:03:50.000000000 +1000
```
```suggestion
--- a/Lib/platform.py
```

Rechi commented on tools/depends/target/python3/darwin_embedded.patch:
```diff
@@ -0,0 +1,276 @@
+         except OSError as e:
+diff -ru Python-3.7.4_original/Lib/platform.py Python-3.7.4/Lib/platform.py
+--- Python-3.7.4_original/Lib/platform.py	2019-07-09 04:03:50.000000000 +1000
++++ Python-3.7.4/Lib/platform.py	2019-08-21 16:15:46.000000000 +1000
```
```suggestion
+++ b/Lib/platform.py
```

Rechi commented on tools/depends/target/python3/darwin_embedded.patch:
```diff
@@ -0,0 +1,269 @@
+--- a/Lib/os.py	2019-07-09 04:03:50.000000000 +1000
```
```suggestion
--- a/Lib/os.py
```

Rechi commented on tools/depends/target/python3/darwin_embedded.patch:
```diff
@@ -0,0 +1,269 @@
+--- a/Lib/os.py	2019-07-09 04:03:50.000000000 +1000
++++ b/Lib/os.py	2019-08-21 10:58:34.000000000 +1000
```
```suggestion
+++ b/Lib/os.py
```

Rechi commented on tools/depends/target/python3/darwin_embedded.patch:
```diff
@@ -0,0 +1,269 @@
+         except (FileNotFoundError, NotADirectoryError) as e:
+             last_exc = e
+         except OSError as e:
+--- a/Lib/platform.py	2019-07-09 04:03:50.000000000 +1000
```
```suggestion
--- a/Lib/platform.py
```

Rechi commented on tools/depends/target/python3/darwin_embedded.patch:
```diff
@@ -0,0 +1,269 @@
+             last_exc = e
+         except OSError as e:
+--- a/Lib/platform.py	2019-07-09 04:03:50.000000000 +1000
++++ b/Lib/platform.py	2019-08-21 16:15:46.000000000 +1000
```
```suggestion
+++ b/Lib/platform.py
```

fuzzard commented:
jenkins build this please

fuzzard committed:
[ios] fix runtime python3

_posixsubprocess.c cant be not compiled, as the os module automatically adds
subprocess that requires _posixsubprocess.c.
